assigner/roster_util.py

```diff
@@ -16,25 +16,26 @@ def get_filtered_roster(roster, section, target):
     return roster
 
 
-def add_to_roster(conf, backend, roster, name, username, section, force=False):
+def add_to_roster(
+    conf, backend, roster, name, username, section, force=False, canvas_id=-1
```

Comment:
Can we use `None` here rather than `-1` to signal not having an id? That would better match `typing`'s idea of `Optional`, I think.